Doc/whatsnew/3.14.rst

```diff
@@ -118,6 +118,10 @@ Others
   are removed. They had previously raised a :exc:`DeprecationWarning`
   since Python 3.12.
 
+* The *type*, *choices*, and *metavar* parameters
+  of :class:`!argparse.BooleanOptionalAction` are removed.
+  They were deprecated since 3.12.
```

Comment:
Please can you move this into an "argparse" section, above the "Others" I added in https://github.com/python/cpython/pull/118799? Let's list the modules alphabetically like in 3.13.